### PR TITLE
[argtable3] Update to v3.2.1

### LINKED ISSUE #12297: [argtable3] spurious build failure
**Host Environment**
- OS: Windows Server 2019
- Compiler: Visual Studio 2019 version 16.6

**To Reproduce**
`./vcpkg install argtable3:x64-uwp` (presumably)

**Failure logs** 
[argtable3.zip](https://github.com/microsoft/vcpkg/files/4882074/argtable3.zip)

**Additional context**
This failure appears to be spurious, there are likely races in the package's build system:

```
       "D:\buildtrees\argtable3\x64-uwp-dbg\install.vcxproj" (default target) (1) ->
       "D:\buildtrees\argtable3\x64-uwp-dbg\ALL_BUILD.vcxproj" (default target) (3) ->
       "D:\buildtrees\argtable3\x64-uwp-dbg\examples\ls.vcxproj" (default target) (7) ->
       (CustomBuild target) -> 
         CUSTOMBUILD : CMake error : Cannot restore timestamp D:/buildtrees/argtable3/x64-uwp-dbg/examples/CMakeFiles/generate.stamp [D:\buildtrees\argtable3\x64-uwp-dbg\examples\ls.vcxproj]
```

### PR DESCRIPTION
**Pull request by tomghuang**
We have modified Argtable3 CMake scripts, which now respect `BUILD_SHARED_LIBS` and build either the static library or the dynamic library at a time, so the latest Argtable3 can support vcpkg manifest without any issue. In the patch, we not only update the port to the latest stable release, but also remove unnecessary code snippets and deprecated targets/variables.

**Describe the pull request**

- #### What does your PR fix?  
  Fixes #12297. However, this patch is not only used to resolve old issues, but also used to point to the latest stable release.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All triplets should be supported. No, we didn't modify `ci.baseline.txt`.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes, we have followed the maintainer guide.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes, we have run the command to update the version information. You can see `baseline.json` and `argtable3.json` in the patch.

### CODE REVIEW

File: ports/argtable3/portfile.cmake [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
+)
+
+vcpkg_extract_source_archive_ex(
     OUT_SOURCE_PATH SOURCE_PATH
-    REPO argtable/argtable3
-    REF 1c1bb23b305c8cf349328fc0cacd7beb7a575ff4 # v3.1.5
-    SHA512 13150c8adc1eda107b6df65a2e276510a66bd912f6067d7cc72951735a4c20307144b04cda959cdd24f160da3810ba8bb35e48992ff4281e44ed2331d030fb1d
-    HEAD_REF master
+    ARCHIVE ${ARCHIVE}
 )
```
Comment (PhoebeHui) [inline ports/argtable3/portfile.cmake:10]: We prefer to use vcpkg_from_github, could you update it with following code?

```
vcpkg_from_github(
    OUT_SOURCE_PATH SOURCE_PATH
    REPO argtable/argtable3
    REF 52f24e50e1802e7afeb8ee7d090fca1bf86b49c0 # v3.2.1.52f24e5
    SHA512 b1b4ee0b50216b5b190dd2d7c8968d1f0334a878366f597d99606bc4ac09b8628f143eb6f188da8cdd0de1c473ed2650e4b702ac1c1058793f4b0c305031219e
    HEAD_REF master
)
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@tomghuang, thanks for your PR!

File: ports/argtable3/portfile.cmake (hunk 1 above)
Comment (tomghuang) [reply ports/argtable3/portfile.cmake:10]: Hi @PhoebeHui , I'd like to use `vcpkg_from_github()`. However, I don't want to download the source tar.gz file prepared by GitHub. I want to download the release archive (`argtable-v3.2.1.52f24e5.tar.gz` in this case) prepared by myself. That's why I use `vcpkg_extract_source_archive_ex()` instead.

The archive prepared by GitHub *doesn't* have the `version.tag` file in the root directory. This file is critical for preparing `Argtable3ConfigVersion.cmake` and other version-related CMake functionality. That's why I want vcpkg using the release archives prepare by ourselves.

Is it possible to ask `vcpkg_from_github()` to download a specific asset of a release? Thanks.

Comment (PhoebeHui) [reply ports/argtable3/portfile.cmake:10]: > The archive prepared by GitHub doesn't have the version.tag file in the root directory.

@tomghuang, in this case, I think using vcpkg_download_distfile is fine, thanks for clarifying this!

**Review by PhoebeHui — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Great, this is awesome! Thanks @tomghuang :)

**Pull request merged**